metadata/src/test/java/org/apache/kafka/metalog/LocalLogManager.java

```diff
@@ -227,14 +228,14 @@ synchronized long tryAppend(int nodeId, int epoch, List<ApiMessageAndVersion> ba
 
         synchronized long tryAppend(int nodeId, int epoch, LocalBatch batch) {
             if (epoch != leader.epoch()) {
-                log.trace("tryAppend(nodeId={}, epoch={}): the provided epoch does not " +
+                log.debug("tryAppend(nodeId={}, epoch={}): the provided epoch does not " +
                     "match the current leader epoch of {}.", nodeId, epoch, leader.epoch());
-                return Long.MAX_VALUE;
+                throw new NotLeaderException("Append failed because the replication is not the current leader");
```

Comment:
Looks like `KafkaRaftClient` throws `IllegalArgumentException` when the passed `epoch` is greater than the current leader epoch. This is based on `BatchAccumulator`:
```java
          if (epoch < this.epoch) {
              throw new NotLeaderException("Append failed because the epoch doesn't match");
          } else if (epoch > this.epoch) {
              throw new IllegalArgumentException("Attempt to append from epoch " + epoch +
                  " which is larger than the current epoch " + this.epoch);
          }
```

Reply:
Thanks for pointing this out. I have updated LocalLogManager to adhere to the same.